src/components/VmDetails/cards/SnapshotsCard/SnapshotItem.js

```diff
@@ -41,110 +41,154 @@ SnapshotAction.propTypes = {
-      />)
-
-      // Delete action
-      buttons.push(<DeleteConfirmationModal
-        key='delete'
-        disabled={isActionsDisabled}
-        id={`${id}-delete-modal`}
-        trigger={
-          <SnapshotAction key='delete' id={`${id}-delete`}>
-            <OverlayTrigger placement='left' overlay={<Tooltip id={`${id}-delete-tt`}>{ msg.snapshotDelete() }</Tooltip>}>
-              <Icon type='pf' name='delete' />
-            </OverlayTrigger>
-          </SnapshotAction>
-        }
-        onDelete={onSnapshotDelete}
-      >
-        <div
-          dangerouslySetInnerHTML={{
-            __html: msg.areYouSureYouWantToDeleteSnapshot({
-              snapshotName: `"<strong>${escapeHtml(snapshot.get('description'))}</strong>"`,
-            }),
-          }}
-        />
-        <div>{msg.thisOperationCantBeUndone()}</div>
-      </DeleteConfirmationModal>)
+    this.updateScreenType = this.updateScreenType.bind(this)
+  }
+
+  componentDidUpdate () {
+    this.updateScreenType()
+  }
+
+  updateScreenType () {
+    const state = { isMobile: false, isTablet: false }
+    if (window.innerWidth <= 768 && window.innerWidth > 600) {
+      state.isMobile = false
+      state.isTablet = true
+    } else if (window.innerWidth <= 600) {
+      state.isMobile = true
+      state.isTablet = false
+    } else {
+      state.isMobile = false
+      state.isTablet = false
     }
 
-    // Status tooltip
-    const tooltipId = `${id}-status-icon-${snapshot.get('status')}`
-    switch (snapshot.get('status')) {
-      case 'locked':
-        statusIcon = <StatusTooltip icon={<Icon type='pf' name='locked' />} text={msg.locked()} id={tooltipId} />
-        break
-      case 'in_preview':
-        statusIcon = <StatusTooltip icon={<Icon type='fa' name='eye' />} text={msg.inPreview()} id={tooltipId} />
-        break
-      case 'ok':
-        statusIcon = <StatusTooltip icon={<Icon type='pf' name='ok' />} text={msg.ok()} id={tooltipId} />
-        break
+    if (this.state.isMobile !== state.isMobile || this.state.isTablet !== state.isTablet) {
+      this.setState(state)
     }
   }
 
-  return (
-    <div className={style['snapshot-item']} id={id}>
-      <span className={style['snapshot-item-status']} id={`${id}-status-icon`}>{statusIcon}</span>
-      <span className={style['snapshot-item-name']} id={`${id}-description`}>
-        {getMinimizedString(snapshot.get('description'), MAX_DESCRIPTION_SIZE)}
-        <span className={style['snapshot-item-time']} id={`${id}-time`}>{`(${formatDateFromNow(snapshot.get('date'))})`}</span>
-      </span>
-      <span className={style['snapshot-item-actions']} id={`${id}-actions`}>{ buttons }</span>
-    </div>
-  )
+  componentDidMount () {
+    window.addEventListener('resize', this.updateScreenType)
+    this.updateScreenType()
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('resize', this.updateScreenType)
+  }
+
+  render () {
+    const StatusTooltip = ({ icon, text, id }) => {
```

Comment:
Please, move this component outside of `SnapshotItem`

Reply:
@bond95 This component is here because its tooltip placement is always left. I can place this outside, then add a prop to it.

Reply:
@thaorell yes, please, do this. Because placing component into another component it is anti-pattern. Component must have only one way of input, it is props. This was done to make debug easier and code more clear.

Reply:
@bond95 Thx for the feedbacks! I've already finished this part on the latest commit. take a look :)